src/main/java/net/dries007/tfc/client/gui/overlay/PlayerDataOverlay.java

```diff
@@ -0,0 +1,178 @@
+package net.dries007.tfc.client.gui.overlay;
+
+import java.awt.*;
+
+import org.lwjgl.opengl.GL11;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.ScaledResolution;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.GuiIngameForge;
+import net.minecraftforge.client.event.RenderGameOverlayEvent;
+import net.minecraftforge.client.event.RenderGameOverlayEvent.ElementType;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import net.dries007.tfc.util.DamageManager;
+
+import static net.dries007.tfc.api.util.TFCConstants.MOD_ID;
+
+public final class PlayerDataOverlay
+{
+    private static final ResourceLocation ICONS = new ResourceLocation(MOD_ID, "textures/gui/overlay/icons.png");
+    private static final PlayerDataOverlay INSTANCE = new PlayerDataOverlay();
+
+    public static PlayerDataOverlay getInstance() { return INSTANCE; }
+    private FontRenderer fontrenderer = null;
+    private float maxHealth = 1000, curThirst = 100;
+
+    private PlayerDataOverlay() {}
+
+    public void setMaxHealth(float value) { this.maxHealth = value; }
```

Comment:
These should just read directly from the capability. This is potential for errors since there's no guarantee that these will be called when the capability gets updated

Reply:
Looks like I missed changing this back after I fixed the problem I had at the early implementation of the overlay(it wasn't reading capabilities). Fixed in the new commit.